src/FamilyHubs.Referral.Web/Pages/ProfessionalReferral/SupportDetails.cshtml.cs

```diff
@@ -0,0 +1,74 @@
+using FamilyHubs.Referral.Core.Models;
+using FamilyHubs.Referral.Core.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace FamilyHubs.Referral.Web.Pages.ProfessionalReferral;
+
+public class SupportDetailsModel : PageModel
+{
+    private readonly IDistributedCacheService _distributedCacheService;
+
+    public string ServiceId { get; private set; } = default!;
+    public string ServiceName { get; private set; } = default!;
+
+    public PartialTextBoxViewModel PartialTextBoxViewModel { get; set; } = new PartialTextBoxViewModel()
+    {
+        HeadingText = "Who should the service contact in the family?",
+        HintText = "This must be a person aged 16 or over.",
+        TextBoxLabel = "Full name",
+        MainErrorText = "Enter a full name",
+        TextBoxErrorText = "Enter a full name",
+    };
+
+
+    [BindProperty]
+    public string TextBoxValue { get; set; } = string.Empty;
+
+    public SupportDetailsModel(IDistributedCacheService distributedCacheService)
+    {
+        _distributedCacheService = distributedCacheService;
+    }
+
+    public void OnGet(string serviceId, string serviceName)
+    {
+        //Fixes Session Changing between requests 
+        this.HttpContext.Session.Set("What", new byte[] { 1, 2, 3, 4, 5 });
```

Comment:
Could you please help me understand the issue that this code fixes?

Reply:
If you don't add this line of code, then between the OnGet and OnPost the session id changes rendering anything you try to store and subsequently retrieve completely useless